staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -288,12 +291,22 @@ func (o AnnotateOptions) RunAnnotate() error {
 		}
 
 		var outputObj runtime.Object
+		var dataChangeMsg string
 		obj := info.Object
 
 		if o.dryRunStrategy == cmdutil.DryRunClient || o.local || o.list {
+			oldData, err := json.Marshal(obj)
```

Comment:
Is there any other way to discriminate `annotate` and `unannotate` without marshaling objects?

Reply:
I wonder can't we extract this "modified/unmodified" value from `updateAnnotations` function?. Same question also for not dry-run cases.